[FEAT] 플라이웨이 적용, test db docker compose 작성, 환경별 application.yml 분리

park-se-jun opened the pull request:
## PR Description 🗒️
- [x] 플라이웨이 적용
- [x] test db docker compose 파일 작성
- [x] 환경별 application yml 분리
- [x] RDS 연결 <- prod application.yml 작성 및 cd workflow, deploy.sh 수정 하긴 했는데 제대로 될지 잘 모르겠음
## 추가/변경 사항 및 설명 📝
- application.yml 의 profile을 로컬환경에서의 local, test환경에서의 test, 배포 환경에서의 prod(아직 못만듦)로 분리
- db 형상관리 툴인 플라이웨이 적용
  - V1_init.sql 부터 시작해서, Entity의 변경사항이 있을때마다 그 변경사항에 대한 sql 문을 새로 작성해 버전을 같이 올려주면서 관리 해야함
## To Reviewers 🙏
localTestDB에 db_info.env 파일이 아래와 같은 형식으로 존재해야 합니다.
```
MYSQL_DATABASE= 
MYSQL_USER=
MYSQL_PASSWORD=
MYSQL_ROOT_PASSWORD=
```
괜찮다면, 일시적으로 pr에 cd를 걸어놔도 괜찮을까요??

## Issue close ✂️
close #41

park-se-jun committed:
[CHORE] local용 db를 실행 시키는 docker compose 작성

park-se-jun committed:
[FEAT] flyway 적용 및 로컬 환경에서의 db 연결

park-se-jun committed:
[CHORE] application.yml 환경별 분리
- test 환경의 경우 h2사용하도록 분리
- local 환경의 경우 localhost:3306을 사용하도록 분리
- prod 환경일 경우 아직 작성 못함

park-se-jun committed:
[CHORE] prod 관련 applicatoin.yml 작성 및 cd 수정
- application-prod.yml 작성
- deploy.sh 에 profile 관련 옵션 추가
- deploy workflow에 prod의 db 관련 정보 설정

park-se-jun committed:
Merge pull request #57 from prgrms-web-devcourse/feature_#41

[FEAT] 플라이웨이 적용, test db docker compose 작성, 환경별 application.yml 분리

thguss reviewed (approved):
수고하셨습니다~!
잘 몰라서 그런건지는 모르겠지만 우선 수정할 사항은 없을 것 같고, 테스트 커버리지 때문에 테스트 작업이 develop에 push 된 후 상화 보고 merge 하면 좋을 것 같아요! 또한, env 파일도 공유해주시면 감사하겠습니다. 이 점은 슬랙에서 다시 언급할게요~!

twotwobread commented on src/main/resources/db/migration/V1__init.sql:
```diff
@@ -0,0 +1,113 @@
+create table battle
```
이 초기 버전이 있고 DB 구조가 변하거나 컬럼이 추가되거나 할때 해당 구조로 바꾸기 위한 sql을 추가로 날리는 그런 방법이 맞을까요??

park-se-jun replied:
entity가 수정 될때마다 그에 맞는 DDL문을 db/migration에 새로운 버전으로 (ex] V2__updateBattle.sql) 과 같이 추가하면 됩니다. 만약entity와 db가 일치하지 않는다면, 실행시 에러가 발생합니다.

twotwobread reviewed (approved):
남긴 코멘트는 궁금해서 물어본거라서 approve 하겠습니다.

park-se-jun committed:
Merge branch 'develop' into sejun

park-se-jun committed:
[CHORE] 임시로 pr과 deploy.yml 연결

park-se-jun committed:
[CHORE] pr 과 deploy.yml 연결 제거, 테스트용 db pr-check.yml에 추가

park-se-jun committed:
[CHORE] test시 test profile을 이용하도록 build 설정
- docker compose 관련 워크플로 삭제
- test 시 test profile 을 이용하도록 변경

park-se-jun committed:
[CHORE] pr에 deploy.yml 연결
- deploy.yml 트리거로 pr 연결
- pr-check 의 이름을 pr-check로 변경(원래 develop이었음)

park-se-jun committed:
[CHORE] shell script 수정 - profile을 prod로

park-se-jun committed:
[CHORE] 배포는 되지만 실행이 되지 않는 문제
- application-prod.yml 에 env를 file path로 import 하는데, file path는 커맨드 실행하는데를 기준으로 해서 배포시 잡지 못하는 문제가 있었음
- deploy.sh 에 -Dspring.config.import 옵션으로 덮어쓰기

park-se-jun committed:
[CHORE] spring.config.import 를 classpath로 변경한거에서 filepath로 되돌리기
- shell 파일 수정을 통해, 절대경로로 prod_db_info를 받게끔 했는데 application야믈 에는 classpath로 되어 잇었

park-se-jun committed:
[CHORE] deploy.yml에 임시로 pr 트리거 걸어 뒀던 것 제

thguss reviewed (approved):
수고하셨습니다!

twotwobread reviewed (approved):
수고하셨습니다!!
